src/Kestrel.Transport.Sockets/Internal/CoalescingScheduler.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.Internal
+{
+    public class CoalescingScheduler
```

Comment:
Should we rename this? It isn't s PipeScheduler (why isn't it a pipe scheduler)?